src/VolumeLogger.js

```diff
@@ -59,6 +59,295 @@ module.exports = class VolumeLogger {
+                                    resolve(messages)
+                                }
+                            }
+                        )
+                    }
+                })
+            }
+
+            const minReport = {
+                peerName: '',
+                broker: {
+                    messagesToNetworkPerSec: 0,
+                    bytesToNetworkPerSec: 0,
+                    messagesFromNetworkPerSec: 0,
+                    bytesFromNetworkPerSec: 0,
+                },
+                network: {
+                    avgLatencyMs: 0,
+                    bytesToPeersPerSec: 0,
+                    bytesFromPeersPerSec: 0,
+                    connections: 0,
+                },
+                storage: {
+                    bytesWrittenPerSec: 0,
+                    bytesReadPerSec: 0,
+                },
+
+                startTime: 0,
+                currentTime: 0,
+                timestamp: 0
+            }
+
+            const metricsReportFn = async () => {
+                // ...
+                try {
+                    sec += 1
+
+                    const metricsReport = await this.metricsContext.report()
+                    // console.log(metricsReport)
+                    // console.log(metricsReport.metrics)//.WsEndpoint.msgSpeed)
+
+                    const secReport = {
+                        // peerId: brokerAddress,
+                        peerName: metricsReport.peerId,
+                        broker: {
+                            messagesToNetworkPerSec: metricsReport.metrics['broker/publisher'].messages.rate,
+                            bytesToNetworkPerSec: metricsReport.metrics['broker/publisher'].bytes.rate,
+                            messagesFromNetworkPerSec: 0,
+                            bytesFromNetworkPerSec: 0,
+                        },
+                        network: {
+                            avgLatencyMs: metricsReport.metrics.node.latency.rate,
+                            bytesToPeersPerSec: metricsReport.metrics.WebRtcEndpoint.outSpeed.rate || 0,
+                            bytesFromPeersPerSec: metricsReport.metrics.WebRtcEndpoint.inSpeed.rate || 0,
+                            connections: metricsReport.metrics.WebRtcEndpoint.connections || 0
+                        },
+                        storage: {
+                            bytesWrittenPerSec: (metricsReport.metrics['broker/cassandra']) ? metricsReport.metrics['broker/cassandra'].writeBytes : 0,
+                            bytesReadPerSec: (metricsReport.metrics['broker/cassandra']) ? metricsReport.metrics['broker/cassandra'].readBytes : 0,
+                        },
+
+                        startTime: metricsReport.startTime,
+                        currentTime: metricsReport.currentTime,
+                        timestamp: metricsReport.currentTime,
+
+                    }
+
+                    const hourReport = {
```

Comment:
If I'm understanding this correctly, the initialization of variables `secReport`, `hourReport` and `dayReport` is repeated code. Just create once and do a deep copy?